barretenberg/cpp/src/barretenberg/stdlib/primitives/memory/ram_table.cpp

```diff
@@ -75,78 +77,67 @@ template <typename Builder> ram_table<Builder>::ram_table(const std::vector<fiel
  *
  * @details initialize the table once we perform a read. This ensures we always have a pointer to a Builder.
  * (if both the table entries and the index are constant, we don't need a builder as we
- * can directly extract the desired value fram `_raw_entries`)
+ * can directly extract the desired value fram `raw_entries`)
  *
  * @tparam Builder
  */
 template <typename Builder> void ram_table<Builder>::initialize_table() const
 {
-    if (_ram_table_generated_in_builder) {
+    if (ram_table_generated_in_builder) {
         return;
     }
-    BB_ASSERT(_context != nullptr);
-    _ram_id = _context->create_RAM_array(_length);
+    // only call this when there is a context
+    BB_ASSERT(context != nullptr);
+    ram_id = context->create_RAM_array(length);
 
-    if (_raw_entries.size() > 0) {
-        for (size_t i = 0; i < _length; ++i) {
-            if (!_index_initialized[i]) {
+    if (raw_entries.size() > 0) {
+        for (size_t i = 0; i < length; ++i) {
+            if (!index_initialized[i]) {
                 field_pt entry;
-                if (_raw_entries[i].is_constant()) {
-                    entry = field_pt::from_witness_index(_context,
-                                                         _context->put_constant_variable(_raw_entries[i].get_value()));
+                if (raw_entries[i].is_constant()) {
+                    entry = field_pt::from_witness_index(context,
+                                                         context->put_constant_variable(raw_entries[i].get_value()));
                 } else {
-                    entry = _raw_entries[i];
+                    entry = raw_entries[i];
                 }
-                _context->init_RAM_element(_ram_id, i, entry.get_witness_index());
-                _index_initialized[i] = true;
+                context->init_RAM_element(ram_id, i, entry.get_witness_index());
+                index_initialized[i] = true;
             }
         }
     }
 
     // Store the tags of the original entries
-    _tags.resize(_length);
-    if (_raw_entries.size() > 0) {
-        for (size_t i = 0; i < _length; i++) {
-            _tags[i] = _raw_entries[i].get_origin_tag();
+    _tags.resize(length);
+    if (raw_entries.size() > 0) {
+        for (size_t i = 0; i < length; i++) {
+            _tags[i] = raw_entries[i].get_origin_tag();
         }
     }
-    _ram_table_generated_in_builder = true;
+    ram_table_generated_in_builder = true;
 }
 
-/**
- * @brief Construct a new ram table<Builder>::ram table object
- *
- * @tparam Builder
- * @param other
- */
 template <typename Builder>
 ram_table<Builder>::ram_table(const ram_table& other)
```

Comment:
Looks to me like all of these constructors/operator= can just be defaulted. Signifies intent and avoids simple errors like the one you corrected below